GitHubTasks IsGitHubRepository throws an exception if the GitRepository does not have any defined remotes.

### Usage Information

6.3.0

### Description

GitHubTasks IsGitHubRepository method throws a NullReferenceException when a repository does not have any defined remotes.

This leads to the CodeGenerationTask failing, although I suspect there would be a number of other locations where the error would likely appear.

```
 error MSB4018: The "CodeGenerationTask" task failed unexpectedly.
 error MSB4018: System.NullReferenceException: Object reference not set to an instance of an object.
 error MSB4018:    at Nuke.Common.Tools.GitHub.GitHubTasks.IsGitHubRepository(GitRepository repository) in /_/source/Nuke.Common/Tools/GitHub/GitHubTasks.cs:line 118
 error MSB4018:    at Nuke.MSBuildTasks.CodeGenerationTask.<>c__DisplayClass20_0.<ExecuteInner>g__GetSourceFile|4(Tool tool) in /_/source/Nuke.MSBuildTasks/CodeGenerationTask.cs:line 53
```

### Reproduction Steps

Perform a task that uses IsGitHubRepository on a repository that does not have any remotes.

### Expected Behavior

The exception should not be thrown.

### Actual Behavior

An NullReferenceException is thrown.

### Regression?

_No response_

### Known Workarounds

Add a remote to the repository.

### Could you help with a pull-request?

Yes

--- a/source/Nuke.Common/Tools/GitHub/GitHubTasks.cs
+++ b/source/Nuke.Common/Tools/GitHub/GitHubTasks.cs
@@ -125,7 +125,7 @@ public static class GitHubTasks
 
     public static bool IsGitHubRepository(this GitRepository repository)
     {
-        return repository != null && repository.Endpoint.EqualsOrdinalIgnoreCase("github.com");
+        return repository?.Endpoint?.EqualsOrdinalIgnoreCase("github.com") ?? false;
     }
 
     public static string GetGitHubOwner(this GitRepository repository)

